plugin/trino-hive/src/main/java/io/trino/plugin/hive/aws/athena/projection/DateProjectionFactory.java

```diff
@@ -0,0 +1,206 @@
+    private static final Pattern DATE_RANGE_BOUND_EXPRESSION_PATTERN = Pattern.compile("^\\s*NOW\\s*(([+-])\\s*([0-9]+)\\s*(DAY|HOUR|MINUTE|SECOND)S?\\s*)?$");
+
+    @Override
+    public boolean isSupportedColumnType(Type columnType)
+    {
+        return columnType instanceof VarcharType
+                || columnType instanceof DateType
+                || (columnType instanceof TimestampType && ((TimestampType) columnType).isShort());
+    }
+
+    @Override
+    public Projection create(String columnName, Type columnType, Map<String, Object> columnProperties)
+    {
+        String dateFormatPattern = getProjectionPropertyRequiredValue(
+                columnName,
+                columnProperties,
+                COLUMN_PROJECTION_FORMAT,
+                String::valueOf);
+
+        List<String> range = getProjectionPropertyRequiredValue(
+                columnName,
+                columnProperties,
+                COLUMN_PROJECTION_RANGE,
+                value -> ((List<?>) value).stream()
+                        .map(String.class::cast)
+                        .collect(toImmutableList()));
+        if (range.size() != 2) {
+            throw invalidRangeProperty(columnName, dateFormatPattern);
+        }
+
+        SimpleDateFormat dateFormat = new SimpleDateFormat(dateFormatPattern);
+        dateFormat.setLenient(false);
+        dateFormat.setTimeZone(TimeZone.getTimeZone(UTC_TIME_ZONE_ID));
+        Supplier<Instant> leftBound = parseDateRangerBound(columnName, range.get(0), dateFormat);
+        Supplier<Instant> rangeBound = parseDateRangerBound(columnName, range.get(1), dateFormat);
+        if (!leftBound.get().isBefore(rangeBound.get())) {
+            throw invalidRangeProperty(columnName, dateFormatPattern);
+        }
+
+        int interval = getProjectionPropertyValue(columnProperties, COLUMN_PROJECTION_INTERVAL, Integer.class::cast).orElse(1);
+        ChronoUnit intervalUnit = getProjectionPropertyValue(columnProperties, COLUMN_PROJECTION_INTERVAL_UNIT, ChronoUnit.class::cast)
+                .orElseGet(() -> resolveDefaultChronoUnit(columnName, dateFormatPattern));
+
+        if (!DATE_PROJECTION_INTERVAL_UNITS.contains(intervalUnit)) {
+            throw invalidProjectionException(
+                    columnName,
+                    format(
+                            "Property: '%s' value '%s' is invalid. Available options: %s",
+                            COLUMN_PROJECTION_INTERVAL_UNIT,
+                            intervalUnit,
+                            DATE_PROJECTION_INTERVAL_UNITS));
+        }
+
+        return new DateProjection(columnName, dateFormat, leftBound, rangeBound, interval, intervalUnit);
+    }
+
+    private ChronoUnit resolveDefaultChronoUnit(String columnName, String dateFormatPattern)
+    {
+        String datePatternWithoutText = dateFormatPattern.replaceAll("'.*'", "");
```

Comment:
Will not work as '.*' is greedy. Also the quoted block can have escaped `'` inside. I think you cannot work around it using regular expression (those have to little power).
Can we just reject any patterns with `'` inside?

Reply:
Oh - actually there are no escapes. To put a single `'` in pattern you use `''`.
So you can filter out quoted section with regex. Just make `.*` non-greedy: `"'.*?'"`

Reply:
And add a test :)

Reply:
Added